src/device.rs

```diff
@@ -547,25 +548,73 @@ impl<B: UsbBus> UsbDevice<'_, B> {
                 Ok(())
             }),
 
-            descriptor_type::STRING => {
-                if index == 0 {
-                    accept_writer(xfer, |w| {
-                        w.write(descriptor_type::STRING, &lang_id::ENGLISH_US.to_le_bytes())
-                    })
-                } else {
-                    let s = match index {
-                        1 => config.manufacturer,
-                        2 => config.product,
-                        3 => config.serial_number,
-                        _ => {
-                            let index = StringIndex::new(index);
-                            let lang_id = req.index;
+            descriptor_type::STRING => match index {
+                // first STRING Request
+                0 => {
+                    if let Some(extra_lang_ids) = config.extra_lang_ids {
+                        assert!(
+                            extra_lang_ids.len() < 16,
+                            "Not support more than 15 extra LangIDs"
+                        );
+
+                        let mut lang_id_bytes = [0u8; 32];
+                        lang_id_bytes[0..2].copy_from_slice(&lang_id::ENGLISH_US.to_le_bytes());
+
+                        extra_lang_ids
+                            .iter()
+                            .enumerate()
+                            .for_each(|(index, lang_id)| {
+                                lang_id_bytes[(index + 1) * 2..(index + 2) * 2]
+                                    .copy_from_slice(&lang_id.to_le_bytes());
+                            });
```

Comment:
This is much simpler by using rust's fancy iterator logic. You may need to clean this up a bit to make it valid rust, but it should look as follows:

```suggestion
                        let mut lang_id_bytes = [0u8; 32];

                        for (buffer, lang_id) in lang_id_bytes.chunks_exact_mut(2).zip(&[lang_id::ENGLISH_US].iter().chain(extra_lang_ids.iter())) {
                            buffer.copy_from_slice(&lang_id.to_le_bytes());
                        }
```